src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/DomainEventServiceTest.kt

```diff
@@ -132,5 +128,22 @@ class DomainEventServiceTest {
         }
       )
     }
+
+    verify(exactly = 1) {
+      mockHmppsTopic.snsClient.publish(
+        match {
+          val deserializedMessage = objectMapper.readValue(it.message, SnsEvent::class.java)
+
+          deserializedMessage.eventType == "approved-premises.application.submitted" &&
+            deserializedMessage.version == 1 &&
+            deserializedMessage.description == "An application has been submitted for an Approved Premises placement" &&
+            deserializedMessage.detailUrl == domainEventToSave.data.eventDetails.applicationUrl &&
+            deserializedMessage.occurredAt == domainEventToSave.occurredAt &&
+            deserializedMessage.additionalInformation.applicationId == applicationId &&
+            deserializedMessage.personReference.identifiers.any { it.type == "CRN" && it.value == domainEventToSave.data.eventDetails.personReference.crn } &&
+            deserializedMessage.personReference.identifiers.any { it.type == "NOMS" && it.value == domainEventToSave.data.eventDetails.personReference.noms }
+        }
+      )
+    }
```

Comment:
I'm thinking that it's probably quite important to demonstrate that the Domain Event Service does NOT emit an event if it's unable to `save()` the event? 

Should we have an additional scenario where `domainEventRespositoryMock.save()` returns something unhappy or raises an error and then assert that `mockHmppsTopic.snsClient.publish()` received no messages?

Reply:
Good shout, I've added a test for this case too